src/js/media-controller.js

```diff
@@ -937,27 +943,33 @@ const monitorForMediaStateReceivers = (
 };
 
 let testMediaEl;
-export const getTestMediaEl = (nodeName = "video") => {
+export const getTestMediaEl = (nodeName = 'video') => {
   if (testMediaEl) return testMediaEl;
-  if (typeof window !== "undefined") {
-    testMediaEl = document.createElement(nodeName);
-  }
+  testMediaEl = document?.createElement?.(nodeName);
   return testMediaEl;
 };
 
-export const hasVolumeSupportAsync = async (
-  mediaEl = getTestMediaEl()
-) => {
+export const hasVolumeSupportAsync = async (mediaEl = getTestMediaEl()) => {
   if (!mediaEl) return false;
   const prevVolume = mediaEl.volume;
   mediaEl.volume = prevVolume / 2 + 0.1;
-  return new Promise((resolve, _reject) => {
-    setTimeout(() => {
-      resolve(mediaEl.volume !== prevVolume);
-    }, 0);
+  return Promise.resolve().then(() => {
+    mediaEl.volume !== prevVolume;
   });
 };
 
+export const hasPipSupport = (mediaEl = getTestMediaEl()) =>
+  typeof mediaEl?.requestPictureInPicture === 'function';
```

Comment:
We should probably use this inside of the pipUnavailable getter, so that a media-controller with no media can have a smart default, but then if a media that doesn't have the pip API exposed is added it can update correctly.